src/components/AttachmentModal.js

```diff
@@ -42,28 +45,91 @@ const propTypes = {
+
         this.state = {
-            isModalOpen: false,
-            file: null,
-            sourceURL: props.sourceURL,
+            title: 'Attachment',
         };
     }
 
+    componentDidUpdate(prevProps) {
+        const {visibleAttachment: {isModalOpen}} = this.props;
+        if (!prevProps.visibleAttachment.isModalOpen && isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    // Using regex to get the src attribute from the html img tag
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    // A function to get the next attachment based on the current attachment being show in the report.
+    getNextAttachment = (toRight) => {
+        const {sortedReportActions, visibleAttachment: {sourceURL}} = this.props;
+        const attachments = sortedReportActions.filter(sortedReportAction => sortedReportAction.action.isAttachment);
+        if (attachments.length <= 1) {
+            return;
+        }
+
+        // Finding the current attachment inside the report using the sourceURL
+        const currentAttachment = attachments.find((attachment) => {
+            const html = lodashGet(attachment, ['action', 'message', 0, 'html'], '');
+            return html.includes(sourceURL);
+        });
+        const actions = [...sortedReportActions].reverse();
+        const sequenceNumber = currentAttachment.action.sequenceNumber;
+        const actionsToSearch = toRight
+            ? actions.slice(sequenceNumber)
+            : actions.slice(0, sequenceNumber !== 0 ? sequenceNumber - 1 : 0).reverse();
+        let nextAttachment = actionsToSearch.find(actionResult => actionResult.action.isAttachment);
+
+        // If there is no next attachment in the left or right direction,
+        // then we need to loop to get the next attachment after the end of the array
+        if (!nextAttachment) {
+            const actionsToSearchFromBeginning = toRight ? actions : [...actions].reverse();
+            nextAttachment = actionsToSearchFromBeginning.find(actionResult => actionResult.action.isAttachment);
+        }
+        const html = lodashGet(nextAttachment, ['action', 'message', 0, 'html'], '');
+        const attachmentSource = this.getAttachmentSource(html);
+        setAttachmentModalData({
+            sourceURL: attachmentSource, isAttachment: true, isModalOpen: true, file: null,
+        });
+    }
+
+    // Function used to handle Left and Right key presses.
+    // This is a seperate function so that it can be subscribed/unsubscribed from the events
+    handleKeyPress = (event) => {
+        if (event.key === 'ArrowRight' || event.key === 'ArrowLeft') {
```

Comment:
We have an existing library that we can use to bind these events called [`KeyboardShortcut`](https://github.com/Expensify/Expensify.cash/blob/master/src/libs/KeyboardShortcut/index.js)